File: frontend/test-app.js
import { chromium } from '@playwright/test';
import * as fs from 'fs';
import * as path from 'path';

const screenshots = [];

async function runTest() {
  const browser = await chromium.launch();
  const context = await browser.createBrowserContext();
  const page = await context.newPage();

  try {
    console.log('Navigating to http://localhost:5173...');
    await page.goto('http://localhost:5173', { waitUntil: 'networkidle' });

    console.log('Taking screenshot of login page...');
    const loginScreenshot = '/tmp/01-login-page.png';
    await page.screenshot({ path: loginScreenshot, fullPage: true });
    screenshots.push({ name: '01-login-page.png', path: loginScreenshot });
    console.log('Login page screenshot saved:', loginScreenshot);

    console.log('Attempting to log in with admin@example.com / password...');

    // Find and fill email field
    const emailInput = page.locator('input[type="email"]') || page.locator('input[placeholder*="email" i]');
    await emailInput.fill('admin@example.com');
    console.log('Entered email');

    // Find and fill password field
    const passwordInput = page.locator('input[type="password"]');
    await passwordInput.fill('password');
    console.log('Entered password');

    // Find and click login button
    const loginButton = page.locator('button:has-text("Login")') || page.locator('button:has-text("Sign in")') || page.locator('button:has-text("Submit")');
    await loginButton.click();
    console.log('Clicked login button');

    // Wait for navigation to complete
    await page.waitForLoadState('networkidle');
    console.log('Logged in successfully');

    // Take dashboard screenshot
    const dashboardScreenshot = '/tmp/02-admin-dashboard.png';
    await page.screenshot({ path: dashboardScreenshot, fullPage: true });
    screenshots.push({ name: '02-admin-dashboard.png', path: dashboardScreenshot });
    console.log('Dashboard screenshot saved:', dashboardScreenshot);

    // Get page title to verify we're on dashboard
    const title = await page.title();
    console.log('Current page title:', title);

    // Find all navigation links
    const navLinks = await page.locator('nav a, [role="navigation"] a, aside a').all();
    console.log(`Found ${navLinks.length} navigation links`);

    // Test navigation
    for (let i = 0; i < Math.min(navLinks.length, 5); i++) {
      try {
        const link = navLinks[i];
        const href = await link.getAttribute('href');
        const text = await link.textContent();
        console.log(`\nTesting navigation link ${i + 1}: "${text}" (${href})`);

        // Click the link
        await link.click();
        await page.waitForLoadState('networkidle');
        console.log(`Successfully navigated to: ${page.url()}`);

        // Take screenshot of the section
        const sectionScreenshot = `/tmp/0${3 + i}-section-${i + 1}.png`;
        await page.screenshot({ path: sectionScreenshot, fullPage: true });
        screenshots.push({ name: `0${3 + i}-section-${i + 1}.png`, path: sectionScreenshot });
        console.log(`Section screenshot saved: ${sectionScreenshot}`);

        // Go back to avoid issues with navigation state
        await page.goBack();
        await page.waitForLoadState('networkidle');
      } catch (e) {
        console.log(`Error testing link ${i + 1}:`, e.message);
      }
    }

    console.log('\n=== Test Summary ===');
    console.log('Screenshots taken:');
    screenshots.forEach((s, idx) => {
      console.log(`${idx + 1}. ${s.name}: ${s.path}`);
    });
    console.log('All tests completed successfully!');

  } catch (error) {
    console.error('Error during test:', error);
    process.exit(1);
  } finally {
    await browser.close();
  }
}

runTest();

File: frontend/test-app.py
#!/usr/bin/env python3
import time
import sys
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.chrome.service import Service

screenshots = []

def take_screenshot(driver, name, description):
    path = f"/tmp/{name}"
    driver.save_screenshot(path)
    screenshots.append((name, path, description))
    print(f"Screenshot saved: {name} - {description}")

try:
    # Setup Chrome driver
    print("Initializing Chrome driver...")
    chrome_options = Options()
    chrome_options.add_argument("--headless")
    chrome_options.add_argument("--no-sandbox")
    chrome_options.add_argument("--disable-dev-shm-usage")
    chrome_options.add_argument("--disable-gpu")

    # Try using the system chromium directly
    from selenium.webdriver.chrome.service import Service as ChromeService
    try:
        driver = webdriver.Chrome(options=chrome_options)
    except:
        service = Service(ChromeDriverManager(version="148").install())
        driver = webdriver.Chrome(service=service, options=chrome_options)

    # Navigate to login page
    print("\n1. Navigating to http://localhost:5173...")
    driver.get("http://localhost:5173")
    time.sleep(3)

    # Take login page screenshot
    take_screenshot(driver, "01-login-page.png", "Login page")

    # Verify we're on login page
    page_source = driver.page_source
    if "login" in page_source.lower() or "email" in page_source.lower():
        print("✓ Login page loaded successfully")

    # Attempt login
    print("\n2. Logging in with admin@example.com / password...")
    try:
        # Find email input
        email_input = WebDriverWait(driver, 10).until(
            EC.presence_of_element_located((By.CSS_SELECTOR, "input[type='email'], input[placeholder*='email' i], input[name*='email' i]"))
        )
        email_input.send_keys("admin@example.com")
        print("✓ Entered email address")

        # Find password input
        password_input = driver.find_element(By.CSS_SELECTOR, "input[type='password']")
        password_input.send_keys("password")
        print("✓ Entered password")

        # Find and click login button
        login_button = driver.find_element(By.CSS_SELECTOR, "button:contains('Login'), button:contains('Sign in'), button:contains('Submit'), button[type='submit']")
        login_button.click()
        print("✓ Clicked login button")

    except:
        # Alternative method if selectors fail
        print("Trying alternative login method...")
        inputs = driver.find_elements(By.TAG_NAME, "input")
        if len(inputs) >= 2:
            inputs[0].send_keys("admin@example.com")
            inputs[1].send_keys("password")
            buttons = driver.find_elements(By.TAG_NAME, "button")
            for btn in buttons:
                if btn.text.lower() in ["login", "sign in", "submit"]:
                    btn.click()
                    break

    # Wait for navigation
    time.sleep(3)

    # Take dashboard screenshot
    take_screenshot(driver, "02-admin-dashboard.png", "Admin dashboard after login")

    # Verify we're logged in
    current_url = driver.current_url
    page_source = driver.page_source
    print(f"✓ Current URL: {current_url}")

    # Test navigation
    print("\n3. Testing navigation through different sections...")

    # Find all navigation links
    nav_links = []
    try:
        nav_elements = driver.find_elements(By.CSS_SELECTOR, "nav a, [role='navigation'] a, aside a, [class*='nav'] a, [class*='menu'] a")
        nav_links = [(elem.text, elem.get_attribute('href')) for elem in nav_elements if elem.text.strip()]
        nav_links = list(dict.fromkeys(nav_links))  # Remove duplicates
        nav_links = nav_links[:5]  # Limit to 5 links
    except:
        print("Could not find navigation elements")

    print(f"Found {len(nav_links)} navigation links")

    for idx, (link_text, link_href) in enumerate(nav_links, 1):
        if not link_text.strip():
            continue
        print(f"\n  Testing link {idx}: '{link_text.strip()}'")
        try:
            driver.get(f"http://localhost:5173{link_href}" if link_href.startswith("/") else link_href if link_href else "http://localhost:5173")
            time.sleep(2)
            take_screenshot(driver, f"0{3+idx}-section-{idx}.png", f"Section: {link_text.strip()}")
            print(f"  ✓ Successfully navigated to section")
        except Exception as e:
            print(f"  ✗ Error: {str(e)}")

    # Print summary
    print("\n" + "="*50)
    print("TEST SUMMARY")
    print("="*50)
    print(f"Application tested successfully!")
    print(f"\nScreenshots captured:")
    for idx, (name, path, desc) in enumerate(screenshots, 1):
        print(f"  {idx}. {name} - {desc}")
        print(f"     Path: {path}")

    driver.quit()
    print("\nTest completed successfully!")

except Exception as e:
    print(f"ERROR: {str(e)}")
    import traceback
    traceback.print_exc()
    sys.exit(1)
